sharding/handler/smc_handler.py

```diff
@@ -34,25 +35,33 @@
 smc_json = get_smc_json()
 
 
-class SMCHandler(Contract):
+class SMC(Contract):
 
-    logger = logging.getLogger("evm.chain.sharding.SMCHandler")
+    logger = logging.getLogger("evm.chain.sharding.SMC")
     abi = smc_json["abi"]
     bytecode = decode_hex(smc_json["bytecode"])
 
-    private_key = None  # type: datatypes.PrivateKey
-    sender_address = None  # type: Address
+    default_priv_key = None  # type: datatypes.PrivateKey
+    default_sender_address = None  # type: Address
     config = None  # type: Dict[str, Any]
 
+    _estimate_gas_dict = dict()  # type: Dict[str, Any]
```

Comment:
As it's the same for all instances, we could populate it here already instead of in `__init__` (or even outside the class and only reference it here).

Reply:
And the type should be `Dict[str, int]` instead of `Dict[str, Any]`.